### PR TITLE
update to use py3 syntax

### PR DESCRIPTION
**Pull request by xiangyan99**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/core/azure-core/azure/core/polling/base_polling.py [hunk 1]
```diff
@@ -198,15 +198,15 @@ class OperationResourcePolling(LongRunningOperation):
     """
 
     def __init__(
-        self, operation_location_header="operation-location", **kwargs
+        self, operation_location_header="operation-location", *, lro_options={}
```
Comment (lmazuel) [inline sdk/core/azure-core/azure/core/polling/base_polling.py:201]: Never default to a dict, ever :)

**Review by lmazuel — CHANGES_REQUESTED** [review]
Default to dict is an anti-pattern in Python, default to None.

File: sdk/core/azure-core/azure/core/polling/base_polling.py (hunk 1 above)
Comment (xiangyan99) [reply sdk/core/azure-core/azure/core/polling/base_polling.py:201]: This is our original code:

kwargs.pop("lro_options", {}) or {}

Comment (xiangyan99) [reply sdk/core/azure-core/azure/core/polling/base_polling.py]: Updated.

**Review by lmazuel — APPROVED** [review]

**Review by annatisch — APPROVED** [review]

**Pull request merged**